bazaarvoice/emodb
# Fixed issue with databus request forwarding

## Github Issue #

[69](https://github.com/bazaarvoice/emodb/issues/69)

## What Are We Doing Here?

This pull request resolves the issue with databus resource forwarding.  It does so by replacing `DatabusClientSubjectProxy` with `SubjectDatabus`.  The key difference is that `SubjectDatabus` directly mimics the `AuthDatabus` interface and therefore can correctly route traffic by use of `@PartitionKey` annotations.

## How to Test and Verify

1. Check out this PR
2. Run multiple EmoDB servers or multiple local instances using different ports while sharing the same Cassandra and ZooKeeper processes.
3. Create multiple subscriptions with different names.
4. Poll each subscription on each server.  The one with is the partition for the subscription should respond locally.  This can be verified by a message such as "com.bazaarvoice.emodb.event.owner.OstrichOwnerGroup: Starting owned service databus-dedup: subscription-name" in the logs.  All other servers should forward the request to the owned service partition.  This can be verified by watching the request logs.

## Risk

Since databus request forwarding is already not working properly the only risk is that some other functionality is broken entirely.

### Level 

`Medium`

### Required Testing

`Regression`

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
